datasets/iq/aml_list/crawler.py

```diff
@@ -0,0 +1,135 @@
+import re
+from openpyxl import load_workbook
+from rigour.mime.types import XLSX
+from typing import Optional
+
+from zavod import Context, helpers as h
+
+SOURCE_URLS = ["https://aml.iq/?page_id=2169", "https://aml.iq/?page_id=2171"]
+
+LOC_PERSON_LISTS = [
+    "2017",
+    "2018",
+    "2019",
+    "2020",
+    "2021",
+    "2022",
+    "2023",
+    "2024",
+    "2025",
+]
+LOC_ENT_LIST = ["الكيانات"]
+LOC_IGNORE_LIST = ["الافراد", "القوائم المحلية "]
+INT_LISTS = ["كيانات", "افراد"]
+
+YEAR_PATTERN = re.compile(r"\b(" + "|".join(LOC_PERSON_LISTS) + r")\b")
+ENTITY_NAME_REASON = re.compile(r"\s*للتوسط ببيع وشراء العملات الاجنبية$")
+
+
+def clean_entity_name(entity_name: str) -> Optional[str]:
+    if entity_name:
+        match = ENTITY_NAME_REASON.search(entity_name)
+        if match:
+            return ENTITY_NAME_REASON.sub("", entity_name).strip()
+    return entity_name
+
+
+def extract_reason(entity_name: str) -> Optional[str]:
+    """Extracts the reason for listing from the entity name."""
+    if entity_name is None:
+        return None
+    match = ENTITY_NAME_REASON.search(entity_name)
+    return match.group().strip() if match else None
+
+
+def extract_listing_date(decision_number: str) -> Optional[str]:
+    """Extracts the listing year from the decision number."""
+    if decision_number is None:
+        return None
+    match = YEAR_PATTERN.search(decision_number)
+    return match.group(0) if match else None
+
+
+def crawl_row(row: dict, context: Context):
+    entity_name = row.pop("asm_alkyan", None)  # اسم الكيان
```

Comment:
could you pass a header_lookup arg to parse_xlsx_sheet to translate these to English?